File: DAL/Entities/Permission.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace DAL.Entities
{
    public class Permission
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public Guid Id { get; set; } = new();
        public Role? Role { get; set; }
        public Guid RoleId { get; set; }
        public required string ControllerName { get; set; }
        public bool IsView { get; set; }
        public bool IsAdd { get; set; }
        public bool IsEdit { get; set; }
        public bool IsDelete { get; set; }
        public bool IsApprove { get; set; }
        public bool IsStatistic { get; set; }
    }
}

File: CLIENT/src/app/models/system/models/permission.ts
export interface Permission {
  role: string | null;
  roleId: string | null;
  controllerName: string | null;
  isView: boolean;
  isAdd: boolean;
  isEdit: boolean;
  isDelete: boolean;
  isApprove: boolean;
  isStatistic: boolean;
}
